### PR TITLE
[MINI-3453] Feat/dynamic deeplink

### PR DESCRIPTION
**Pull request by rleojoseph**
# Description
* Add custom deep links via **Settings->General->Deeplinks-> +**
* Any deeplink that user has entered in the settings, Sample app will try to launch them using `UIApplication.shared.open(requestURL, options: [:], completionHandler: nil)`

**NOTE:**
* These deeplinks are not supported in External Webview now

## Links
[MINI-3453](https://jira.rakuten-it.com/jira/browse/MINI-3453)

# Checklist
- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [ ] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `fastlane ci` without errors

### CODE REVIEW

File: MiniApp/Classes/core/MiniAppSdkConfig.swift [hunk 1]
```diff
@@ -108,5 +117,6 @@ public class MiniAppSdkConfig {
         self.analyticsConfigList = analyticsConfigList
         self.requireMiniAppSignatureVerification = requireMiniAppSignatureVerification
         self.sslKeyHash = sslKeyHash
+        self.dynamicDeepLinksList = dynamicDeepLinksList
```
Comment (khairul-alam-licon) [inline MiniApp/Classes/core/MiniAppSdkConfig.swift:120]: Do you think, we can pass `dynamicDeepLinksList` using `MiniApp.create` instead to keep a similar experience with `queryParams`?

File: Example/Models/UserProfileModel.swift [hunk 2]
```diff
@@ -67,6 +83,10 @@ func updateContactList(list: [MAContact]?) {
     }
 }
 
+func updateDeeplinkList(list: [String]?) {
+    _ = setDeepLinksList(deeplinksList: list)
```
Comment (Climbatize) [inline Example/Models/UserProfileModel.swift:87]: You should set the the result as discardable (`@discardableResult`)
```suggestion
    setDeepLinksList(deeplinksList: list)
```

File: Example/Models/UserProfileModel.swift [hunk 3]
```diff
@@ -51,6 +51,22 @@ func getProfileSettings(key: String = "UserProfileDetail") -> UserProfileModel?
...
 
+func getDeepLinksList(key: String = "DeeplinkList") -> [String] {
+    if let deeplinksData = UserDefaults.standard.data(forKey: key) {
+        let deeplinksList = try? PropertyListDecoder().decode([String].self, from: deeplinksData)
+        return deeplinksList ?? []
+    }
+    return []
+}
+
+func setDeepLinksList(forKey key: String = "DeeplinkList", deeplinksList: [String]? = getDeepLinksList()) -> Bool {
```
Comment (Climbatize) [inline Example/Models/UserProfileModel.swift:62]:
```suggestion
@discardableResult func setDeepLinksList(forKey key: String = "DeeplinkList", deeplinksList: [String]? = getDeepLinksList()) -> Bool {
```

File: MiniApp/Classes/core/MiniAppSdkConfig.swift [hunk 4]
```diff
@@ -81,6 +81,14 @@ public class MiniAppSdkConfig {
         }
     }
 
+    public var dynamicDeepLinksList: [String]? {
+        didSet {
+            if dynamicDeepLinksList?.count == 0 {
```
Comment (Climbatize) [inline MiniApp/Classes/core/MiniAppSdkConfig.swift:86]: This should not give the expected result, you should try this:

```suggestion
            if dynamicDeepLinksList?.count ?? 0 == 0 {
```

or 

```suggestion
            if dynamicDeepLinksList == nil {
```

**Review by Climbatize — COMMENTED** [review]
The PR looked very good, too bad you have to refactor it, I still give you my suggestions in case you need to move some pieces of code

**Pull request merged**